src/test/basics/base_uint_test.cpp

```diff
@@ -102,9 +126,56 @@ struct base_uint_test : beast::unit_test::suite
         n = beast::zero;
         BEAST_EXPECT(n == z);
 
-        auto hash = sha512Half(u, v, z, n);
-        BEAST_EXPECT(to_string(hash) ==
-            "55CAB39C72F2572057114B4732210605AA22C6E89243FBB5F9943130D813F902");
+        test96 x { (--test96 { z }) ^ (++test96 { z }) };
+        uset.insert(x);
+        BEAST_EXPECT(to_string(x) == "FFFFFFFFFFFFFFFFFFFFFFFE");
+
+        BEAST_EXPECT(uset.size() == 4);
+
+        // SetHex tests...
+        test96 fromHex;
+        BEAST_EXPECT(fromHex.SetHexExact(to_string(u)));
+        BEAST_EXPECT(fromHex == u);
+        fromHex = z;
+
+        // fails with extra char
+        BEAST_EXPECT(! fromHex.SetHexExact("A" + to_string(u)));
+        BEAST_EXPECT(fromHex != u);
+        fromHex = z;
+
+        // fails with extra char at end, but the value is still parsed (?)
```

Comment:
The design of `base_uint` does not follow best practices. It should mimic the interface of `int` (and its associated free functions in `std::`). Outside the scope of this pull request, but something we will eventually want to address.